src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetection.swift

```diff
@@ -42,54 +69,30 @@ final class ExposureDetection {
 		}
 	}
 
-	private func detectExposureWindows(writtenPackages: WrittenPackages, exposureConfiguration: ENExposureConfiguration) {
+	private func detectExposureWindows(_ keyPackageDownload: KeyPackageDownloadProtocol, writtenPackages: WrittenPackages, exposureConfiguration: ENExposureConfiguration) {
 		if progress != nil {
 			Log.error("previous running progress found, will try to cancel", log: .riskDetection)
 			progress?.cancel()
 		}
 		progress = delegate?.detectExposureWindows(
 			self,
 			detectSummaryWithConfiguration: exposureConfiguration,
-			writtenPackages: writtenPackages
-		) { [weak self] result in
-			writtenPackages.cleanUp()
-			self?.useExposureWindowsResult(result)
-		}
-	}
-
-	private func useExposureWindowsResult(_ result: Result<[ENExposureWindow], Error>) {
-		switch result {
-		case .success(let exposureWindows):
-			didDetectExposureWindows(exposureWindows)
-		case .failure(let error):
-			endPrematurely(reason: .noExposureWindows(error))
-		}
-	}
-
-	typealias Completion = (Result<[ENExposureWindow], DidEndPrematurelyReason>) -> Void
-
-    func start(completion: @escaping Completion) {
-        self.completion = completion
-
-        Log.info("ExposureDetection: Start writing packages to file system.", log: .riskDetection)
-
-        writeKeyPackagesToFileSystem { [weak self] writtenPackages in
-            guard let self = self else { return }
-
-            Log.info("ExposureDetection: Completed writing packages to file system.", log: .riskDetection)
-
-			if !self.deviceTimeCheck.isDeviceTimeCorrect {
-				Log.warning("ExposureDetection: Detecting exposure windows skipped due to wrong device time.", log: .riskDetection)
-				self.endPrematurely(reason: .wrongDeviceTime)
-			} else {
-				Log.info("ExposureDetection: Start detecting exposure windows.", log: .riskDetection)
-				let exposureConfiguration = ENExposureConfiguration(from: appConfiguration.exposureConfiguration)
-				self.detectExposureWindows(writtenPackages: writtenPackages, exposureConfiguration: exposureConfiguration)
+			writtenPackages: writtenPackages,
+			completion: { [weak self] result in
+				defer {
+					writtenPackages.cleanUp()
+				}
+
+				switch result {
+				case .success(let exposureWindows):
+					keyPackageDownload.markPackagesAsCheckedForExposures(writtenPackages.fingerprints)
```

Comment:
I think its only needed to pass the fingerprints from the packages of type keys.

Reply:
I added a filter to the fingerprints property in WrittenPackages